src/aliases.js

```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const writeFileAtomic = require('write-file-atomic');
+const writeFileAtomicSync = writeFileAtomic.sync;
+
+const ALIASES_FILE = {directory: './settings', fileName: './settings/aliases.json'}
+
+const defaultAliases = {
+    add: ["!add"],
+    back: ["!back"],
+    brb: ["!brb", "!lurk"],
+    clear: ["!clear"],
+    close: ["!close"],
+    current: ["!current"],
+    customcode: ["!customcode", "!customcodes"],
+    dismiss: ["!dismiss", "!skip", "!complete", "!completed"],
+    level: ["!level"],
+    list: ["!list", "!queue"],
+    modnext: ["!modnext"],
+    modrandom: ["!modrandom"],
+    next: ["!next"],
+    open: ["!open"],
+    order: ["!order"],
+    pause: ["!pause"],
+    persistence: ["!persistence"],
+    position: ["!position", "!pos"],
+    punt: ["!punt"],
+    random: ["!random"],
+    remove: ["!remove", "!leave"],
+    replace: ["!replace", "!change", "!swap"],
+    restart: ["!restart"],
+    resume: ["!resume"],
+    select: ["!select"],
+    start: ["!start"],
+    submitted: ["!submitted", "!entry", "!mylevel", "!mylvl"],
+    subnext: ["!subnext"],
+    subrandom: ["!subrandom"],
+    weightedchance: ["!weightedchance", "!odds", "!chance", "!chances"],
+    weightednext: ["!weightednext"],
+    weightedrandom: ["!weightedrandom"],
+    weightedsubnext: ["!weightedsubnext"],
+    weightedsubrandom: ["!weightedsubrandom"]
+}
+
+let aliases;
+
+const Aliases = {
+    saveAliases : () => {
+        if(!fs.existsSync(ALIASES_FILE.directory)){
+            fs.mkdirSync(ALIASES_FILE.directory);
+        }
+        writeFileAtomicSync(ALIASES_FILE.fileName, JSON.stringify(aliases));
+    },
+    loadAliases : (create = false) => {
+        if(create){
+            const defaults = JSON.stringify(defaultAliases, null, 2);
+            if(!fs.existsSync(ALIASES_FILE.directory)){
+                fs.mkdirSync(ALIASES_FILE.directory, {recursive: true});
+            }
+            writeFileAtomicSync(ALIASES_FILE.fileName, defaults);
+            aliases = defaultAliases;
+        }
+        if(!create && !fs.existsSync(ALIASES_FILE.fileName)){
+            Aliases.loadAliases(true);
+        }
+        try {
+            aliases = JSON.parse(fs.readFileSync(ALIASES_FILE.fileName, { encoding: "utf8" }));
+        } catch (err) {
+            console.warn('An error occurred when trying to load %s. %s', ALIASES_FILE.fileName, err.message);
+            throw err;
+        }
+    },
+    addAlias : (cmd, alias) => {
+        if(!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)){
+            return false;
+        }
+        if(Object.keys(aliases).includes(alias) || Object.values(aliases).includes(alias)){
+            return false;
+        }
```

Comment:
Here `alias` could start with an exclamation mark (`!`) and could also not start with an exclamation mark (`!`).
Maybe `alias` could be set to `"!" + alias` at the beginning of the function if `alias` does not start with an exclamation mark (`!`).

Also if the suggestion with allowing aliases for other commands would be implemented (see <https://github.com/fluid-queue/fluid-queue/pull/24#issuecomment-1315114148>), then checking the `Object.keys(aliases)` would need to be removed.